src/qmapshack/gis/rte/router/CRouterRoutino.cpp

```diff
@@ -248,19 +245,49 @@ void CRouterRoutino::buildDatabaseList()
                 continue;
             }
 
+            // qDebug() << "buildDatabase Prefix" << prefix;
+
 #ifdef Q_OS_WIN
             Routino_Database * data = Routino_LoadDatabase(dir.absolutePath().toLocal8Bit(), prefix.toLocal8Bit());
 #else
             Routino_Database * data = Routino_LoadDatabase(dir.absolutePath().toUtf8(), prefix.toUtf8());
 #endif
-            if(data)
+            qDebug() << "Loaded Routino DB" << dir.absolutePath().toUtf8().data() << "  " << prefix.toUtf8().data();
+
+            if(data == nullptr)
             {
-                comboDatabase->addItem(prefix.replace("_", " "), quint64(data));
+                QMessageBox::critical(this, "Routino Load...", xlateRoutinoError(Routino_errno), QMessageBox::Abort);
+                continue;
             }
-            else
+            /* determine the profile to use for each database*/
+            QVariantMap dmap;
+            dmap["db"] = QVariant ((qulonglong)data);
+
+            /* check possible profiles.xml locations and use the first available */
+            dmap["profilesPath"] = "";
+            QStringList profilesPaths = {
+                dir.filePath(prefix+"-profiles.xml"),
+                dir.filePath("profiles.xml"),
+                setup->routinoPath("profiles.xml").toUtf8()
+            };
+            for(const QString& profilePath : profilesPaths)
             {
-                QMessageBox::critical(this, "Routino...", xlateRoutinoError(Routino_errno), QMessageBox::Abort);
+                QFileInfo pinfo = QFileInfo(profilePath);
+                if( pinfo.isReadable())
+                {
+                    dmap["profilesPath"] = pinfo.filePath();
+                    break;
+                }
+            }
+            if( dmap["profilesPath"].toString().length() == 0)
+            {
+                QMessageBox::critical(this, "Routino...",tr("Could not find a profiles XML file in expected folders. Routing will not function"), QMessageBox::Ok);
```

Comment:
Why I left it like this is that it reports the error but is does not cause any segmentation errors further on - just no action.  I thought that throwing you out of the program is a bit severe.  Other routing options are available without aborting.  Suggestions welcomed.

Reply:
Using QMessageBox::Abort instead of QMessageBox::Ok will not terminate anything. It's simply about naming the button. And aborting to load the particular profile for that particular database is what you are doing. It's also good pracatice to mention the database you are aborting to load and configure.